File: BlaBlaCar.BL/ODT/TripModels/TripModel.cs
﻿using BlaBlaCar.BL.ODT.CarModels;
using BlaBlaCar.DAL.Entities;

namespace BlaBlaCar.BL.ODT.TripModels
{
    public class TripModel
    {
        public Guid Id { get; set; }
        public string StartPlace { get; set; }
        public string EndPlace { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public int PricePerSeat { get; set; }
        public string Description { get; set; }
        public Guid UserId { get; set; }
        public UserModel? User { get; set; }
        public Guid CarId { get; set; }
        public CarModel? Car { get; set; }
        public ICollection<TripUserModel>? TripUsers { get; set; }
        public ICollection<AvailableSeatsModel>? AvailableSeats { get; set; }
    }
}

File: web-client/src/app/interfaces/trip-interfaces/trip-model.ts
import { AvailableSeatsModel } from "./available-seats-model";
import { BookedTripUsersModel } from "./booked-trip-users-model";
import { CarModel } from "../car-interfaces/car-model";
import { TripUserModel } from "./trip-user-model";
import { UserModel } from "../user-interfaces/user-model";
import { UserPermissionsTrip } from "src/app/enums/user-permissions-trip";

export interface TripModel {
    id: number;
    startPlace: string;
    endPlace: string;
    startLat?: number;
    startLon?: number;
    endLat?: number;
    endLon?: number;
    startTime: Date;
    endTime: Date;
    pricePerSeat: number;
    description: string;
    countOfSeats: number;
    userId: string;
    availableSeats: AvailableSeatsModel[];
    car: CarModel
    user?: UserModel;
    tripUsers: any[];
    bookedTripUsers?: BookedTripUsersModel[];
    TotalTrips?: number;
    userPermission?: UserPermissionsTrip;
}
